### PR TITLE
Fix str.Remove(str.Length) to not throw

### LINKED ISSUE #50094: Make string.Remove(length of string) not throw but return this instead
## Background and Motivation

Currently, for example, string.Remove(5) for a string of length 5 throws. For consistency with the other overload of Remove, as well as Substring, this should be changed to not throw but return the unmodified string instead.

The documentation should also be changed or clarified accordingly.

## Proposed API

Change the following in String.Remove(int):
```diff
        public string Remove(int startIndex)
        {
            if (startIndex < 0)
                throw new ArgumentOutOfRangeException(nameof(startIndex), SR.ArgumentOutOfRange_StartIndex);

-            if (startIndex >= Length)
+           if (startIndex > Length)
                throw new ArgumentOutOfRangeException(nameof(startIndex), SR.ArgumentOutOfRange_StartIndexLessThanLength);

            return Substring(0, startIndex);
        }
```
## Usage Examples

``` C#
var str = "abcd";
var test = str.Remove(4);
Debug.Assert(str == test);
```
should not throw an exception.

## Alternative Designs

none

## Risks

One could theoretically depend on the exception being thrown.

### PR DESCRIPTION
**Pull request by stephentoub**
Fixes https://github.com/dotnet/runtime/issues/50094

### CODE REVIEW

**Review by jkotas — APPROVED** [review]

**Review by GrabYourPitchforks — APPROVED** [review]

**Comment by GrabYourPitchforks** [comment]
Short and sweet. 👍

**Comment by danmoseley** [comment]
Strictly, the docs now need an update (the other overload is wrong too)
https://docs.microsoft.com/en-us/dotnet/api/system.string.remove?view=net-5.0#System_String_Remove_System_Int32_

File: src/libraries/System.Private.CoreLib/src/System/String.Manipulation.cs [hunk 1]
```diff
@@ -874,11 +874,8 @@ public string Remove(int startIndex, int count)
         // a remove that just takes a startindex.
         public string Remove(int startIndex)
         {
-            if (startIndex < 0)
-                throw new ArgumentOutOfRangeException(nameof(startIndex), SR.ArgumentOutOfRange_StartIndex);
-
-            if (startIndex >= Length)
-                throw new ArgumentOutOfRangeException(nameof(startIndex), SR.ArgumentOutOfRange_StartIndexLessThanLength);
+            if ((uint)startIndex > Length)
+                throw new ArgumentOutOfRangeException(nameof(startIndex), startIndex < 0 ? SR.ArgumentOutOfRange_StartIndex : SR.ArgumentOutOfRange_StartIndexLessThanLength);
```
Comment (eerhardt) [inline src/libraries/System.Private.CoreLib/src/System/String.Manipulation.cs:878]: > ArgumentOutOfRange_StartIndexLessThanLength

Does this string need to be updated?

https://github.com/dotnet/runtime/blob/6791d05494f5aa4d1a789525c2708e775aafdc07/src/libraries/System.Private.CoreLib/src/Resources/Strings.resx#L1822-L1824

Comment (eerhardt) [reply src/libraries/System.Private.CoreLib/src/System/String.Manipulation.cs:878]: I can stick up a quick PR to fix it, if you agree.

Comment (stephentoub) [reply src/libraries/System.Private.CoreLib/src/System/String.Manipulation.cs:878]: Good catch. Yes, thanks.

**Pull request merged**